### PR TITLE
EVEREST-434 move to errors package

### PR DESCRIPTION
**Pull request by gen1us2k**
[EVEREST-434](https://jira.percona.com/browse/EVEREST-434) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**CHANGE DESCRIPTION**
---
**Problem:**
EVEREST-434

*Short explanation of the problem.*
pkg/errors was deprecated a couple of years ago

**Related pull requests**

- https://github.com/percona/everest-operator/pull/91

**Cause:**
*Short explanation of the root cause of the issue if applicable.*

**Solution:**
*Short explanation of the solution we are providing with this PR.*

The provided solution does not violate Go's standard error handling principles and implements a custom Marshaller to capitalize error messages

**CHECKLIST**
---
**Jira**
- [ ] Is the Jira ticket created and referenced properly?

**Tests**
- [ ] Is an Integration test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?

### CODE REVIEW

**Review by oksana-grishchenko — COMMENTED** [review]
`pkg/errors` still can be found in go.mod and tests 
Screenshot 2023-09-20 at 23 39 53

**Review by recharte — APPROVED** [review]

**Comment by gen1us2k** [comment]
@oksana-grishchenko migrated tests but it's present in mod files because dependencies use it

**Review by oksana-grishchenko — APPROVED** [review]

File: pkg/kubernetes/configs.go [hunk 1]
```diff
@@ -227,10 +227,10 @@ func (k *Kubernetes) updateConfigWithSecret(
 		// rollback the changes
 		_, err := k.UpdateSecret(ctx, oldSecret)
 		if err != nil {
-			k.l.Error(errors.Wrapf(err, "could not revert back secret %s", oldSecret.Name))
+			k.l.Error(errors.Join(err, fmt.Errorf("could not revert back secret %s", oldSecret.Name)))
 		}
 
-		return errors.Wrap(err, "could not update config in Kubernetes")
+		return errors.Join(err, errors.New("could not update config in Kubernetes"))
```
Comment (michal-kralik) [inline pkg/kubernetes/configs.go:233]: We need to update this line. It will always return an error whereas before it returned `nil`.

**Review by michal-kralik — CHANGES_REQUESTED** [review]

File: pkg/kubernetes/configs.go (hunk 1 above)
Comment (michal-kralik) [reply pkg/kubernetes/configs.go:233]: `err` can be `nil` as returned on line 228.
Before it returned either `nil` or a wrapped `err` depending on the value of `err`

**Pull request merged**